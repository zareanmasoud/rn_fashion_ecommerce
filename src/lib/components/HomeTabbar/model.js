// @flow

export type Buttons = {
  press: Function,
  src: string,
  text: string,
};
