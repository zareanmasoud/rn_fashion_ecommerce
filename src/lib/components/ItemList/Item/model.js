// @flow

export type ItemModel = {
  id: string,
  photo: string,
};
