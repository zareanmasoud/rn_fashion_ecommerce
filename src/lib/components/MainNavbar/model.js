// @flow

export type Button = {
  action: Function,
  src: string,
};
