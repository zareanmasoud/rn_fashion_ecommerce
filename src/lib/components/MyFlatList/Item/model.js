// @flow

export type Data = {
  id: string,
  photo: string,
};
