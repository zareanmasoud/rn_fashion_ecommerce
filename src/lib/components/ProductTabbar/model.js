// @flow

export type Specs = {
  action: Function,
  src: string,
  text: string,
};
