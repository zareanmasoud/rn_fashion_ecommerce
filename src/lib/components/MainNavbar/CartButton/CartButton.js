import React from 'react';
import NavbarButton from '../NavbarButton/NavbarButton';
import Badge from '../Badge';
import type {ButtonTwo} from '../model';

type Props = {
  buttonTwo: ButtonTwo,
};

export default function CartButton({buttonTwo}: Props) {
  const {badge} = buttonTwo;
  return (
    <>
      <NavbarButton press={buttonTwo.press} src={buttonTwo.src} />
      <Badge {...badge} />
    </>
  );
}
