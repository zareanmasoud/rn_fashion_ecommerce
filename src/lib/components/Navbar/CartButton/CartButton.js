import React from 'react';
import NavbarButton from '../NavbarButton/NavbarButton';
import Badge from '../Badge';

type Props = {
  spec: any,
};

export default function CartButton({spec}: Props) {
  const {badge} = spec;
  return (
    <>
      <NavbarButton spec={spec} />
      <Badge resources={badge} />
    </>
  );
}
