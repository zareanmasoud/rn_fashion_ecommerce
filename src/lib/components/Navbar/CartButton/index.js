import CartButton from './CartButton';

export {CartButton as default};
