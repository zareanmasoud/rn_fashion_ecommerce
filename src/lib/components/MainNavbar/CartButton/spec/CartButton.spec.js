import React from 'react';
import {shallow} from 'enzyme';
import CartButton from '../CartButton';
import model from './model';

describe('CartButton Component', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(
        <CartButton buttonTwo={model.buttonTwo(0, 'cart_text')} />,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
