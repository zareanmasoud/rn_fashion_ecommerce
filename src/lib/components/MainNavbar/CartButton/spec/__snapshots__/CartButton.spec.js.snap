// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartButton Component Rendering should match to snapshot 1`] = `
<Fragment>
  <NavbarButton
    handlePress={[Function]}
    mode="default"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <Badge
    cart={0}
    cartTextTestID="cart_text"
  />
</Fragment>
`;
